domain/application/service/application_test.go

```diff
@@ -130,7 +128,12 @@ func (s *applicationServiceSuite) TestCreateApplication(c *gc.C) {
 	}
 	s.state.EXPECT().GetModelType(gomock.Any()).Return("caas", nil)
 	s.state.EXPECT().StorageDefaults(gomock.Any()).Return(domainstorage.StorageDefaults{}, nil)
-	s.state.EXPECT().CreateApplication(gomock.Any(), "ubuntu", app, us).Return(id, nil)
+
+	var receivedArgs []application.AddUnitArg
+	s.state.EXPECT().CreateApplication(gomock.Any(), "ubuntu", app, gomock.Any()).DoAndReturn(func(_ context.Context, _ string, _ application.AddApplicationArg, args []application.AddUnitArg) (coreapplication.ID, error) {
+		receivedArgs = args
+		return id, nil
+	})
```

Comment:
Why this change? Surely this is just a far more complex way of saying the same thing?

Also line 133 is very long

Reply:
Because the pointers don't match, mocks equality aren't like `gc.DeepEquals`, they work on ptr equality.